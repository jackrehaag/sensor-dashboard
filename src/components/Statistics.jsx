import React from "react";
import HumidityStat from "./HumidityStat";
import TemperatureStat from "./TemperatureStat";

class Statistics extends React.Component {
  render(props) {
    return (
      <div>
        <div className="ui two statistics">
          <TemperatureStat temp={this.props.lastTemperature} />
          <HumidityStat humidity={this.props.lastHumidity} />
        </div>
        <div className="ui mini center aligned grid statistics">
          <div className="ui orange statistic">
            <div className="value">{this.props.lastMotionDetection}</div>
            <div className="label">
              Last motion detection
              <i className="street view icon" />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Statistics;
