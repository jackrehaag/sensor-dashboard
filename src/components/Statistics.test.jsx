import React from "react";
import Statistics from "./Statistics";
import renderer from "react-test-renderer";

test("it renders", () => {
  const component = renderer.create(
    <Statistics
      lastTemperature={21}
      lastHumidity={50}
      lastMotionDetection="2019-06-21 16:04:02"
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
