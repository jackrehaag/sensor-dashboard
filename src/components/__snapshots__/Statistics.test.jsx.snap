// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div>
  <div
    className="ui two statistics"
  >
    <div
      className="orange current-temp statistic"
    >
      <div
        className="value"
      >
        21
        Â°C
      </div>
      <div
        className="label"
      >
        Temperature 
        <i
          className="thermometer half icon"
        />
      </div>
    </div>
    <div
      className="teal current-temp statistic"
    >
      <div
        className="value"
      >
        50
        %
      </div>
      <div
        className="label"
      >
        Humidity 
        <i
          className="tint icon"
        />
      </div>
    </div>
  </div>
  <div
    className="ui mini center aligned grid statistics"
  >
    <div
      className="ui orange statistic"
    >
      <div
        className="value"
      >
        2019-06-21 16:04:02
      </div>
      <div
        className="label"
      >
        Last motion detection
        <i
          className="street view icon"
        />
      </div>
    </div>
  </div>
</div>
`;
